source/ada/lsp-ada_documents.adb

```diff
@@ -143,13 +157,19 @@ package body LSP.Ada_Documents is
       --  The first line (index 0) starts at offset 1
       Self.Line_To_Index.Append (1);
 
-      for Ind in 1 .. Length (Self.Text) loop
-         if Element (Self.Text, Ind) = Ada.Characters.Wide_Latin_1.LF then
-            --  The contents of Line_To_Index is the first character
-            --  in each line, so index Ind + 1 for the start of line.
-            Self.Line_To_Index.Append (Ind + 1);
-         end if;
+      while J.Forward loop
+         Self.Line_To_Index.Append (Integer (J.First_UTF16_Offset + 1));
+         Last_Line_Terminated := J.Has_Line_Terminator;
       end loop;
+
+      --  XXX Initial implementation depends from index of the line after
+      --  last line terminator, even that this line is not exists actually.
+      --  Thus, append such an index. It will be removed after full switch
+      --  to VSS.
+
+      if Last_Line_Terminated then
+         Self.Line_To_Index.Append (Integer (J.First_UTF16_Offset + 1));
```

Comment:
Perhaps, it's time to replace `Line_To_Index` type definition by replacing `Integer` with `UTF16_OFFset`.

Reply:
It is planned to replace this vector by vector of Markers.